gcp_variant_transforms/beam_io/vcf_header_io.py

```diff
@@ -32,6 +32,9 @@
 from gcp_variant_transforms.beam_io import vcfio
 
 
+BASE_TYPE_KEY = '{}_{}_TYPE'
```

Comment:
Please add a reference to `annotation` in this name, e.g., `BASE_ANNOTATION_TYPE_KEY`, this should make the related changes easier to understand.

Reply:
+1 to be explicit that this new logic is only for annotations; I even think that it's probably better to remove this constant from here, and instead move it inside infer_headers.py (as a private constant), and instead make a public `get_inferred_annotation_type_header_key` that returns the correct string. That way, all the annotation logic is isolated from the generic vcf_header_io, and there is a proper 'api' for getting this key.